projects/storefrontlib/cms-components/cart/validation/cart-validation-warnings-state.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { ReplaySubject, Subscription } from 'rxjs';
+import { take, tap, withLatestFrom } from 'rxjs/operators';
+import { RoutingService, CartModification } from '@spartacus/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CartValidationWarningsStateService implements OnDestroy {
```

Comment:
How about a simpler name for this service?

```suggestion
export class CartValidationStateService implements OnDestroy {
```